From af09f9ba07d96881a72cedab79451c0f37000c1b Mon Sep 17 00:00:00 2001
From: "Thakur, Santosh Krx" <santosh.krx.thakur@intel.com>
Date: Thu, 4 Feb 2016 18:29:06 +0530
Subject: [PATCH 21/24] SoC: Intel: sst: klockwork error in Audio Driver

This patch checks for "NULL" before dereferencing
a pointer

Change-Id: Ie854f87038a98e87f70da0f5f05fa20d496717f6
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18675
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18674
Signed-off-by: Thakur, Santosh Krx <santosh.krx.thakur@intel.com>
Reviewed-on: https://android.intel.com:443/466534
---
 sound/soc/intel/sst/sst_dsp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/sst/sst_dsp.c b/sound/soc/intel/sst/sst_dsp.c
index 818bf81..34800de5 100644
--- a/sound/soc/intel/sst/sst_dsp.c
+++ b/sound/soc/intel/sst/sst_dsp.c
@@ -615,9 +615,10 @@ static int sst_dma_firmware(struct sst_dma *dma, struct sst_sg_list *sg_list)
 		for_each_sg(sg_src_list, sg, length, i) {
 			pr_debug("dma desc %d, length %d\n", i, sg->length);
 			src_addr = sg_phys(sg);
+			if (!sg_dst_list)
+				return -EFAULT;
 			dstn_addr = sg_phys(sg_dst_list);
-			if (sg_dst_list)
-				sg_dst_list = sg_next(sg_dst_list);
+			sg_dst_list = sg_next(sg_dst_list);
 			sst_drv_ctx->desc = dma->ch->device->device_prep_dma_memcpy(
 					dma->ch, dstn_addr, src_addr, sg->length, flag);
 			if (!sst_drv_ctx->desc)
-- 
1.9.1

